agoradatatools/etl/load.py

```diff
@@ -96,52 +123,61 @@ def load(file_path: str, provenance: list, destination: str, syn=None):
         )
 
         print(e)
-        return
+        return None
     except ValueError:
         print(
             "Please make sure that the Synapse id of "
             + "the provenances and the destination are valid"
         )
-        return
+        return None
 
     return (file.id, file.versionNumber)
 
 
-def df_to_json(df: pd.DataFrame, staging_path: str, filename: str):
-    """
-    Converts a data frame into a json file.
-    :param df: a dataframe
-    :param filename: the final file name included in the config file
-    :return: the path of the newly created temporary json file
+def df_to_json(df: pd.DataFrame, staging_path: str, filename: str) -> Union[None, str]:
+    """Converts a data frame into a json file.
+
+    Args:
+        df (pd.DataFrame): DataFrame to be converted to JSON
+        staging_path (str): Path to staging directory
+        filename (str): name of JSON file to be created
+
+    Returns:
+        Union[None, str]: can return None (if the first `try` fails), or a string containing the name of the new JSON file if the function succeeds
     """
 
     try:
         df = df.replace({np.nan: None})
 
         df_as_dict = df.to_dict(orient="records")
 
-        temp_json = open(path.join(staging_path, filename), 'w+')
-        json.dump(df_as_dict, temp_json,
-                  cls=NumpyEncoder,
-                  indent=2)
+        temp_json = open(path.join(staging_path, filename), "w+")
+        json.dump(df_as_dict, temp_json, cls=NumpyEncoder, indent=2)
     except Exception as e:
         print(e)
-        temp_json.close()
+        try:  # this was failing if the `try` above fails before creating `temp_json`
```

Comment:
I noticed that if the `try` failed before the `temp_json` object was initialized, then the `temp_json.close()` line would throw an error since `temp_json` did not exist.

Reply:
Yeah I agree, definitely would simplify things. Should we just open another ticket for that and any other improvements that can link to this PR?

Reply:
Yes, create a ticket to do this work.

Reply:
ticket created [here](https://sagebionetworks.jira.com/browse/IBCDPE-461)